firmware/coins.c

```diff
@@ -28,6 +28,7 @@ const CoinType coins[COINS_COUNT] = {
 	{true, "Litecoin", true, "LTC",  true,  48, true,    1000000, true,   5, false, 0, false, 0, true, "\x19" "Litecoin Signed Message:\n"},
 	{true, "Dogecoin", true, "DOGE", true,  30, true, 1000000000, true,  22, false, 0, false, 0, true, "\x19" "Dogecoin Signed Message:\n"},
 	{true, "Dash",     true, "DASH", true,  76, true,     100000, true,  16, false, 0, false, 0, true, "\x19" "DarkCoin Signed Message:\n"},
+	{true, "Zcash",    true, "ZEC",  true,  65, true,    1000000, true,   5, false, 0, false, 0, true, "\x19" "Zcash Signed Message:\n"},
```

Comment:
What string do you use for signing? Is it really "Zcash Signed Message:\n", if yes then the character preceeding it should not be \x19 (as it is the string length).